### ISSUE TITLE
Add monorepo template

### ISSUE THREAD

**Issue by natemoo-re** [palmerhq/tsdx#122]
### Current Behavior

Currently, users can scaffold out a monorepo manually, then add `tsdx` as a dependency at the root of the project.

### Desired Behavior

Users should be able to scaffold out a monorepo project using `npx tsdx create`. Just like the current templates, users could pass a flag or select from the interactive prompt.

### Suggested Solution

Create two new templates, `monorepo-basic` and `monorepo-react`. In both cases, the only additional dependency would be `lerna`. The root `package.json` scripts would use `lerna run command` and any packages would use `tsdx command`.

1) Add these options directly to the `create` template prompt and add CLI flags for each.

2) Add an additional prompt to the `create` command which allows you to select a project structure (`basic` or `monorepo`?) and add a `—monorepo` CLI flag which will automatically resolve the monorepo templates.

### Who does this impact? Who is this for?

This feature would be very beneficial for anyone managing a larger project who wishes to leverage both `tsdx` and `lerna`.

### Describe alternatives you've considered

If this use case seems out of scope, improving the documentation around integrating with `lerna` would be helpful to guide users through manual setup. 

As a possible alternative, `tsdx` could become monorepo-aware so that running `npx tsdx create mylib` from within a monorepo could scaffold out a new package in the `packages/` (or configured in `lerna.json`) directory.

### Additional context

I have created a minimal [tsdx-monorepo](https://github.com/natemoo-re/tsdx-monorepo) example project.

I would be happy to contribute to this feature, but I want to collect feedback and hash out the preferred approach before starting.

**Comment by jaredpalmer** [comment]
Update: I just moved formik into a monorepo powered by tsdx (with both React native and react packages). Should be a good test bed. We DEF want to bake this in because it was huge pain to setup with VSCode.

**Comment by sw-yx** [comment]
what were the pain points?

**Comment by thaiphan** [comment]
@jaredpalmer Is that something you can blog about?

**Comment by ambroseus** [comment]
@thaiphan  example of using `tsdx` in lerna monorepo : https://github.com/DCKit/dckit

**Comment by mwarger** [comment]
@jaredpalmer - any updates on the baked-in monorepo setup?  Interested to see what you encountered...  Thanks.

**Comment by pcowgill** [comment]
> Update: I just moved formik into a monorepo powered by tsdx (with both React native and react packages). Should be a good test bed. We DEF want to bake this in because it was huge pain to setup with VSCode.

In the meantime, we can look at the `formik` repo as a reference https://github.com/jaredpalmer/formik/

**Comment by kylemh** [comment]
Looks like #778 should cover it and I'm also using TSDX in a lerna monorepo with yarn workspaces [here](https://github.com/AirLabsTeam/air-core).

**Comment by agilgur5** [comment]
@kylemh since #778 doesn't actually add the full workflow with `tsdx create` and replacing naming etc, I wouldn't say this is resolved until then. #778 probably resolves the lesser #275 however.

**Comment by eduardodelaguila** [comment]
I'm trying to setup a mono repo too, i've installed lerna and have installed some dependencies, my question is if the tsdx init would be at root level or at package level? I've been checking @kylemh repo and formik repo but I haven't been able to figure this out.
